src/lib.rs

```diff
@@ -196,6 +232,24 @@ pub trait StructOfArray {
     type Type;
 }
 
+pub trait SoASlice<'a> {
```

Comment:
Could all these traits be made "private", i.e. live inside a non `pub` module? Also, these seems to be used to access the slice type associated with another SoA type when nesting them, right? Could it be possible to generate the corresponding type syntactically, without using Rust type system (something like `field.type.soa_slice_type()`)?

If we have to use traits, could you add a small documentation for users/developers?

Reply:
I removed all the additional trait except `SoAIter`, because all soa iterators are hided behind a detail mod, need a trait to connect them.

Also tried to remove `SoAIter` but declare something like `#field_detail_mod::Iter`, but this could not work if soa types are in different modules.

```rust
mod foo {
   #[derive(StructOfArray)]
   struct Foo {...}
   
   // generated
   mod __detail_iter_foo {
        struct<'a> Iter<'a> {...}
        struct<'a> IterMut<'a> {...}
   }
}
use foo::*;
struct Bar {
    #[nested_soa]
    foo: Foo,
    ...
}
// generated
mod __detail_iter_bar {
    struct<'a> Iter<'a> {
        foo: __detail_iter_foo::Iter<'a> // not work
    }
}

```

Reply:
I added some documentation for `SoAIter`